lib/segment/src/payload_storage/simple_payload_storage.rs

```diff
@@ -175,4 +175,159 @@ mod tests {
+                assert_eq!(x[0], 43);
+            },
+            _ => assert!(false)
+        }
+        match &pload[&"floating".to_string()] {
+            PayloadType::Float(x) => {
+                assert_eq!(x.len(), 1);
+                assert_eq!(x[0], 30.5);
+            },
+            _ => assert!(false)
+        }
+        match &pload[&"boolean".to_string()] {
+            PayloadType::Keyword(x) => {
+                assert_eq!(x.len(), 1);
+                assert_eq!(x[0], "true");
+            },
+            _ => assert!(false)
+        }
+        match &pload[&"metadata__temperature".to_string()] {
+            PayloadType::Float(x) => {
+                assert_eq!(x.len(), 1);
+                assert_eq!(x[0], 60.5);
+            },
+            _ => assert!(false)
+        }
+        match &pload[&"metadata__width".to_string()] {
+            PayloadType::Integer(x) => {
+                assert_eq!(x.len(), 1);
+                assert_eq!(x[0], 60);
+            },
+            _ => assert!(false)
+        }
+        match &pload[&"metadata__height".to_string()] {
+            PayloadType::Integer(x) => {
+                assert_eq!(x.len(), 1);
+                assert_eq!(x[0], 50);
+            },
+            _ => assert!(false)
+        }
+        match &pload[&"metadata__nested__feature".to_string()] {
+            PayloadType::Float(x) => {
+                assert_eq!(x.len(), 1);
+                assert_eq!(x[0], 30.5);
+            },
+            _ => assert!(false)
+        }
+        match &pload[&"string_array".to_string()] {
+            PayloadType::Keyword(x) => {
+                assert_eq!(x.len(), 2);
+                assert_eq!(x[0], "hello");
+                assert_eq!(x[1], "world");
+            },
+            _ => assert!(false)
+        }
+        match &pload[&"integer_array".to_string()] {
+            PayloadType::Integer(x) => {
+                assert_eq!(x.len(), 2);
+                assert_eq!(x[0], 1);
+                assert_eq!(x[1], 2);
+            },
+            _ => assert!(false)
+        }
+        match &pload[&"metadata__integer_array".to_string()] {
+            PayloadType::Integer(x) => {
+                assert_eq!(x.len(), 2);
+                assert_eq!(x[0], 1);
+                assert_eq!(x[1], 2);
+            },
+            _ => assert!(false)
+        }
+        match &pload[&"float_array".to_string()] {
+            PayloadType::Float(x) => {
+                assert_eq!(x.len(), 2);
+                assert_eq!(x[0], 1.0);
+                assert_eq!(x[1], 2.0);
+            },
+            _ => assert!(false)
+        }
+        match &pload[&"boolean_array".to_string()] {
+            PayloadType::Keyword(x) => {
+                assert_eq!(x.len(), 2);
+                assert_eq!(x[0], "true");
+                assert_eq!(x[1], "false");
+            },
+            _ => assert!(false)
+        }
+    }
+
+    #[test]
+    #[should_panic]
```

Comment:
panic in rust could not be handled as exception and leads to immediate program termination. It would be better to use Err result instead of panic in case of unprocessable payload.

Reply:
yes sure, code is not ready. I just wanted to put the thoughts on tha table.

Reply:
this test only pretends to make clear what case is not handled for payloads of arrays with different types

Reply:
If we proceed we could fail the operation or not inserting the values differents to the type of the first element in the Array